c2cgeoportal/views/entry.py

```diff
@@ -70,7 +70,46 @@
 cache_region = get_region()
 
 
-class Entry(object):
+class DimensionInformation:
+
+    URL_PART_RE = re.compile("[a-zA-Z0-9_\-~\.]+$")
+
+    def __init__(self):
+        self.dimensions = {}
+
+    def merge(self, layer, layer_node, mixed):
+        errors = set()
+
+        dimensions = {}
+        for dimension in layer.dimensions:
+            if dimension.value is not None and not self.URL_PART_RE.match(dimension.value):
+                errors.add("The layer '{}' has an unsupported dimension value '{}' ('{}').".format(
+                    layer.name, dimension.value, dimension.name
+                ))
+            elif dimension.name in dimensions:
+                errors.add("The layer '{}' has an duplicated dimension name '{}'.".format(
```

Comment:
OK, I understand better now:
- in this first step you do a validation of the data coming from the database;
- in the second step below you implement the merge with the compatibility rules.

Maybe you should make the dimensions property private and add the last validation in a getter method.